web/src/main/java/org/openmrs/web/filter/initialization/InitializationFilter.java

```diff
@@ -1820,6 +1820,17 @@ public void executing(ChangeSet changeSet, int numChangeSetsToRun) {
 							return;
 						}
 						
+						try {
+							// Update PostgreSQL Sequences after insertion of core data
+							Context.getAdministrationService().updatePostgresSequence();
```

Comment:
Shouldn't we make this call only if we are running PostgreSQL?

Reply:
How about the UpdateFIlter.java?

Reply:
Will this need to be called every time data is inserted?

Reply:
> Shouldn't we make this call only if we are running PostgreSQL?

Yes We are supposed to make the call only when DB is PostgreSQL but since we do not have access to session at this level we cannot determine the DB. Two things come to my mind for this:
1. If it is regarding code readability then we can rename the method to UpdateSequenceIfDbIsPostgreSQL
or
2. We can split this single call into two different calls to AdminService, one two determine if db is PostgreSQL and other to update the sequence if previous call returns true. To me two calls appeared overhead so I went with a single call approach.

Reply:
> Will this need to be called every time data is inserted?

If the data that we are inserting does not holds id values for different entities then this call won't be required. PostgreSQL automatically updates the sequence in this case. But if in the data that is getting inserted we specify the id values for different entities then PostgreSQL does not increment the sequences and then we need to make a call to this service. This is the case [here](https://github.com/openmrs/openmrs-module-referencemetadata/blob/0cbf10b8103b97b9e1e0cd018706c47651ea20ea/api/src/main/java/org/openmrs/module/referencemetadata/ReferenceMetadataActivator.java#L109)

Reply:
> How about the UpdateFIlter.java?

I'll look into it and see if there's a way.

Reply:
> How about the UpdateFIlter.java?

The UpdateFilter is only called when there are insertions via liquibase . But consider [here](https://github.com/openmrs/openmrs-module-referencemetadata/blob/0cbf10b8103b97b9e1e0cd018706c47651ea20ea/api/src/main/java/org/openmrs/module/referencemetadata/ReferenceMetadataActivator.java#L101) where insertions are not via liquibase of Covid-19 Numric Concepts, then we cannot do anything and will get issues. So, I don't think it will work.

Reply:
Could we automate this by making the call from `OpenmrsObjectSaveHandler`?

Reply:
@dkayiwa , OpenMRSObjectSaveHandler is called before the "save__" or "void__", "unvoid_", "retire__", or "unretire___" methods in all the service classes but not around the method [importData](https://github.com/openmrs/openmrs-module-dataexchange/blob/caf00edb7529136581f694640709a3bafb6af3e9/api/src/main/java/org/openmrs/module/dataexchange/DataImporter.java#L56) in dataexhchange module. This method is called to save concepts from the xml dataset files using dbunit (see [here](https://github.com/openmrs/openmrs-module-referencemetadata/blob/0cbf10b8103b97b9e1e0cd018706c47651ea20ea/api/src/main/java/org/openmrs/module/referencemetadata/ReferenceMetadataActivator.java#L97)). 

Also, will it be alright to have the sequences updated even when PostgreSQL updates them automatically ?. There are only two cases where we will want to update the sequences via this service - First, when data is inserted via liquibase and second when a dataset is imported from an xml file ( in these two cases only id of object is specified).